server/src/main/java/org/opensearch/gateway/remote/ClusterMetadataManifest.java

```diff
@@ -613,4 +831,83 @@ public static UploadedIndexMetadata fromXContent(XContentParser parser) throws I
             return PARSER.parse(parser, null);
         }
     }
+
+    /**
+     * Metadata for uploaded metadata attribute
+     *
+     * @opensearch.internal
+     */
+    public static class UploadedMetadataAttribute implements UploadedMetadata, Writeable, ToXContentFragment {
```

Comment:
can we reuse this for IndexMetadata as well by adding support for multiple attribute values instead of 1